### DISCUSSION TITLE
Adding new entities when using pocketbase as a framework

### DISCUSSION THREAD

**Discussion by Mehdi-YC** [pocketbase/pocketbase#2688] [Q&A]
Hello guys , gr8 product btw , it mignt be the django of golang , 
i saw in the documentation this 
https://pocketbase.io/docs/custom-models/
and i wonder , how can i add some models or / collections by default to my pocketbase , 
for example you have Users collection by default
i want for example a collection of cars , how can i make it ? in the documentation you did not talk about saving or registering the model so it will be available in the frontend

**Comment by ganigeorgiev** [comment]
If you are not familiar with Go, I'd recommend using the Admin UI interface to create the collection (_if you enable Automigrate it'll also create the migration file for you_).

If you want to create collection programatically with Go you can check https://pocketbase.io/docs/collection-methods/. Usually they need to be created in a [Migration](https://pocketbase.io/docs/migrations/).

For inserting records programmatically, you can check https://pocketbase.io/docs/record-methods/.

**Reply by Mehdi-YC** [reply]
this is what i tried , created them in the UI and then migrate collections , if this is a clean way i'll do it then , but i need some methods to play well with the default collections , so do i still need to define the new collections in a migration ? or i need a file for them like the article.go example ?

**Reply by ganigeorgiev** [reply]
What do you mean by _"play well with the default collections"_? Could you elaborate what you are trying to do and/or give an example/pseudo-code with the expected workflow?

**Reply by Mehdi-YC** [reply]
imagine that i have a collection named entities , that has an id named collection , a position , a column , and an icon , and is_primary , now what i need is on a creation of a new collection , a record will be created in this collection ,
also i have a collection named field , for each field in the new collections i create , a new record will be created on this one , it has french_name , english_name , icon and other stuff , like metadata for each record (this will be needed for the frontend later)

now for example , to start the fields record creation , i need a trigger post_new_collection_created in the collections collection , where do i write this method , do i create a file collection.go that has the "collections" collection and the methods , or a create a migration for it and just put the handler in the main ?

**Reply by ganigeorgiev** [reply]
I'm sorry, I have difficulties following your example. The naming of your fields is also a little confusing and I don't know what "_has an id named collection_" refers to.

>  to start the fields record creation , i need a trigger post_new_collection_created in the collections collection

But why? What this trigger is supposed to do? Do you mean the [event hooks](https://pocketbase.io/docs/event-hooks/)?

> do i create a file collection.go that has the "collections" collection and the methods , or a create a migration for it and just put the handler in the main ?

You are free to organize your files and packages however you want, but I don't understand the requirements and I'm not sure what to recommend.

**Reply by Mehdi-YC** [reply]
it's a little bit confusing to follow ik , but i'd summarize it in : creation collections of metadata  about the newly created collections
so i'll have a model named collection that has the metadata for all the colelctions , like name fr_name icon position ... this will help me automate the frontend and yes i need hooks , or like we call them signals in django so i need those models to have signals  

now , if i create the models using migrations , where do i define the signals (hooks)
and if i can define the models + it's hooks in a single file how can i do it and add it automatically to pocketbase like the Users collection which is there by default

sorry if it's a little bit confusing and thank you for your answers
